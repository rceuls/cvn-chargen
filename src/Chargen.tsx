import AttributesOverview from './AttributesOverview';
import { generateCharacter } from './services/character.generator';
import CombatStats from './CombatStats';
import SavingThrows from './SavingThrows';
import { getRndValue, getSRDLink } from './services/util';
import { names } from './services/names';
import GenericArrayOverview from './GenericArrayOverview';
import SpellsOverview from './SpellsOverview';

const Chargen = () => {
  const genChar = generateCharacter();
  return (
    <>
      <div className="row">
        <div className="col-sm">
          You are {getRndValue(names)} the{' '}
          <a
            href={getSRDLink(genChar.charClass.url)}
            target="_blank"
            rel="noreferrer"
          >
            {genChar.charClass.name}
          </a>
        </div>
      </div>
      {genChar.charClass.suffix ? (
        <div className="row">
          <div className="col-sm">{genChar.charClass.suffix}</div>
        </div>
      ) : (
        <></>
      )}
      <hr />
      <AttributesOverview character={genChar} />
      <SavingThrows character={genChar} />
      <CombatStats character={genChar} />
      <GenericArrayOverview data={genChar.charClass.abilities} />
      <SpellsOverview data={genChar.charClass.spells ?? []} />
      <GenericArrayOverview data={genChar.charClass.equipment.equipment} />
      <div className="row">
        <div className="col-sm">
          <img
            className="img-thumbnail"
            style={{ maxWidth: 200, maxHeight: 200, aspectRatio: 'auto' }}
            src={`${process.env.PUBLIC_URL}/portraits/${genChar.charClass.image}`}
            alt="your portrait"
          />
        </div>
      </div>
    </>
  );
};

export default Chargen;
