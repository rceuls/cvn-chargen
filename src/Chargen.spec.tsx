import renderer from 'react-test-renderer';
import Chargen from './Chargen';
import { ICharacter } from './services/character.generator';

const genChar: ICharacter = {
  name: 'Tamrin',
  traits: {
    background: 'aaa',
    clothes: 'bbb',
    mannerism: 'ccc',
    personality: 'ddd',
  },
  charClass: {
    bonusXP: (_) => 'my bonus xp',
    name: 'Halfling',
    startingHP: 6,
    savingThrows: [8, 9, 10, 13, 12],
    url: 'Halfling',
    abilities: [
      'Defensive bonus (+2 AC when attacked by larger-than-human opponents)',
      'Hiding (90% in woods, 2-6 otherwise, must remain motionless)',
      'Listen at doors (2-in-6)',
      'Missile attack bonus (+1)',
    ],
    equipment: {
      equipment: [
        'Chain Mail (AC 5)',
        'Polearm (1d10 + STR, two handed, brace, slow)',
        'Backpack',
        'Torches (6)',
        'Tinderbox, flint and steel',
        'Wineskin/Waterskin',
        'Winter Blanket',
        'Rations, Dry, one week',
        'Sack, large',
        'Sacks, small (2)',
        '60 gold pieces',
      ],
      acMod: 4,
    },
    image: 'TCP Human 1.jpg',
  },
  attributeScores: [8, 16, 10, 13, 13, 13],
  maximumHP: 5,
  armorClass: 16,
};

it('renders correctly', () => {
  const tree = renderer.create(<Chargen genChar={genChar} />).toJSON();
  expect(tree).toMatchSnapshot();
});
