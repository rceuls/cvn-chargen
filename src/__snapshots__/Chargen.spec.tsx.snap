// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      You are 
      Tamrin
       the
       
      <a
        href="https://oldschoolessentials.necroticgnome.com/srd/index.php/Halfling"
        rel="noreferrer"
        target="_blank"
      >
        Halfling
      </a>
    </div>
  </div>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-sm-12"
    >
      <span
        className="font-weight-bold"
      >
        STR
      </span>
      <span
        className="ml-2 mr-2"
      >
        8
         
        (-1)
      </span>
      <span
        className="font-weight-bold"
      >
        DEX
      </span>
      <span
        className="ml-2 mr-2"
      >
        16
         
        (+2)
      </span>
      <span
        className="font-weight-bold"
      >
        CON
      </span>
      <span
        className="ml-2 mr-2"
      >
        10
         
      </span>
      <span
        className="font-weight-bold"
      >
        INT
      </span>
      <span
        className="ml-2 mr-2"
      >
        13
         
        (+1)
      </span>
      <span
        className="font-weight-bold"
      >
        WIS
      </span>
      <span
        className="ml-2 mr-2"
      >
        13
         
        (+1)
      </span>
      <span
        className="font-weight-bold"
      >
        CHA
      </span>
      <span
        className="ml-2 mr-2"
      >
        13
         
        (+1)
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-2"
    >
      <span
        className="font-weight-bold"
      >
        Bonus XP
      </span>
      <span
        className="ml-2"
      >
        my bonus xp
      </span>
    </div>
    <div
      className="col-sm-2"
    >
      <span
        className="font-weight-bold"
      >
        Open doors
      </span>
      <span
        className="ml-2"
      >
        1-in-6
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-2"
    >
      <span
        className="font-weight-bold"
      >
        Literacy
      </span>
      <span
        className="ml-2"
      >
        Literate
      </span>
    </div>
    <div
      className="col-sm-10"
    >
      <span
        className="font-weight-bold"
      >
        Languages
      </span>
      <span
        className="ml-2"
      >
        Native + 1 additional
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-2"
    >
      <span
        className="font-weight-bold"
      >
        Max Retainers
      </span>
      <span
        className="ml-2"
      >
        5
      </span>
    </div>
    <div
      className="col-sm-2"
    >
      <span
        className="font-weight-bold"
      >
        Retainer Loyality
      </span>
      <span
        className="ml-2"
      >
        8
      </span>
    </div>
    <div
      className="col-sm-8"
    >
      <span
        className="font-weight-bold"
      >
        NPC reactions
      </span>
      <span
        className="ml-2"
      >
        +1
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-2"
    >
      <span
        className="font-weight-bold"
      >
        Magic Save
      </span>
      <span
        className="ml-2"
      >
        +1
      </span>
    </div>
  </div>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        Death Ray or Poison
      </span>
      <span
        className="ml-2"
      >
        8
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        Magical Wands
      </span>
      <span
        className="ml-2"
      >
        9
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        Paralysis or Petrification
      </span>
      <span
        className="ml-2"
      >
        10
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        Dragon Breath
      </span>
      <span
        className="ml-2"
      >
        13
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        Rods, Staves, Spells
      </span>
      <span
        className="ml-2"
      >
        12
      </span>
    </div>
  </div>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        THAC0 (melee, ranged)
      </span>
      <span
        className="ml-2"
      >
        19 (
        -1
        ,
        +2
        )
      </span>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <span
        className="font-weight-bold"
      >
        Armor Class
      </span>
      <span
        className="ml-2"
      >
        16
      </span>
      <span
        className="ml-2 font-weight-bold"
      >
        Maximum Hit Points
      </span>
      <span
        className="ml-2"
      >
        5
      </span>
    </div>
  </div>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      Listen at doors (2-in-6)
    </div>
    <div
      className="col-sm"
    >
      Defensive bonus (+2 AC when attacked by larger-than-human opponents)
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      Missile attack bonus (+1)
    </div>
    <div
      className="col-sm"
    >
      Hiding (90% in woods, 2-6 otherwise, must remain motionless)
    </div>
  </div>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      Wineskin/Waterskin
    </div>
    <div
      className="col-sm"
    >
      Chain Mail (AC 14)
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      Winter Blanket
    </div>
    <div
      className="col-sm"
    >
      Polearm (1d10 + STR, two handed, brace, slow)
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      Rations, Dry, one week
    </div>
    <div
      className="col-sm"
    >
      Backpack
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      Sack, large
    </div>
    <div
      className="col-sm"
    >
      Torches (6)
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      Sacks, small (2)
    </div>
    <div
      className="col-sm"
    >
      Tinderbox, flint and steel
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      60 gold pieces
    </div>
  </div>,
  <hr />,
  <div
    className="row"
  >
    <div
      className="col-sm"
    >
      <img
        alt="your portrait"
        className="img-thumbnail"
        src="/portraits/TCP Human 1.jpg"
        style={
          Object {
            "aspectRatio": "auto",
            "maxHeight": 200,
            "maxWidth": 200,
          }
        }
      />
    </div>
    <div
      className="col-sm"
    >
      <div
        className="row"
      >
        <div
          className="col-sm"
        >
          Background: 
          aaa
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm"
        >
          Clothes: 
          bbb
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm"
        >
          Personality: 
          ddd
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm"
        >
          Mannerisms: 
          ccc
        </div>
      </div>
    </div>
  </div>,
]
`;
